lib/web_ui/flutter_js/src/skwasm_loader.js

```diff
@@ -5,43 +5,28 @@
 import { createWasmInstantiator } from "./instantiate_wasm.js";
 import { joinPathSegments } from "./utils.js";
 
-export const loadSkwasm = (deps, config, browserEnvironment, baseUrl) => {
-  return new Promise((resolve, reject) => {
-    let skwasmUrl = joinPathSegments(baseUrl, "skwasm.js");
-    if (deps.flutterTT.policy) {
-      skwasmUrl = deps.flutterTT.policy.createScriptURL(skwasmUrl);
-    }
-    const wasmInstantiator = createWasmInstantiator(joinPathSegments(baseUrl, "skwasm.wasm"));
-    const script = document.createElement("script");
-    script.src = skwasmUrl;
-    if (config.nonce) {
-      script.nonce = config.nonce;
-    }
-    script.addEventListener("load", async () => {
-      try {
-        const skwasmInstance = await skwasm({
-          instantiateWasm: wasmInstantiator,
-          locateFile: (fileName, scriptDirectory) => {
-            // When hosted via a CDN or some other url that is not the same
-            // origin as the main script of the page, we will fail to create
-            // a web worker with the .worker.js script. This workaround will
-            // make sure that the worker JS can be loaded regardless of where
-            // it is hosted.
-            const url = scriptDirectory + fileName;
-            if (url.endsWith(".worker.js")) {
-              return URL.createObjectURL(new Blob(
-                [`importScripts("${url}");`],
-                { "type": "application/javascript" }));
-            }
-            return url;
-          }
-        });
-        resolve(skwasmInstance);
-      } catch (e) {
-        reject(e);
+export const loadSkwasm = async (deps, config, browserEnvironment, baseUrl) => {
+  let skwasmUrl = joinPathSegments(baseUrl, "skwasm.js");
+  if (deps.flutterTT.policy) {
+    skwasmUrl = deps.flutterTT.policy.createScriptURL(skwasmUrl);
+  }
+  const wasmInstantiator = createWasmInstantiator(joinPathSegments(baseUrl, "skwasm.wasm"));
+  const skwasm = await import(skwasmUrl);
+  return await skwasm.default({
```

Comment:
(Same comment about the default unnamed export)

Reply:
Same response, emscripten doesn't give us a way to change the export name.